### ISSUE TITLE
OCS-API: how to get users for file sharing

### ISSUE THREAD

**Issue by aosyatnik** [nextcloud/server#15483]
### Steps to reproduce
1. I'm continuing developing [UCS](https://www.univention.com/products/ucs/) application, that uses nextcloud data store. I have a question/issue. My app(as well as nextcloud) is using UCS users. All users(including normal users) should be able to share files with other users.
2. I want to use this api: [**nextcloud_base_url/ocs/v1.php/cloud**](https://docs.nextcloud.com/server/13/admin_manual/configuration_user/user_provisioning_api.html) in order to get available users and groups. But I'm facing the problem: **Logged in user must be at least a sub admin (c) error**, which means, that only admin users can get users and user groups. This is not enough for me! I want to be able to get user groups not only by admin user but also by a normal user. 
3. So, I went even further! I opened nextcloud website, because there I can search for all groups and users as normal user:
![image](https://user-images.githubusercontent.com/8158404/57456908-cc3c8c80-726e-11e9-9d2d-65e543108902.png)

So, I've checked what api nextclod web interface is using. It's using `nextcloud/ocs/v1.php/apps/files_sharing/api/v1/sharees`. I assume this is [version 9](https://docs.nextcloud.com/server/9/developer_manual/core/ocs-share-api.html)

![image](https://user-images.githubusercontent.com/8158404/57457114-2e958d00-726f-11e9-9c3d-021cd292e7b0.png)

And response:
![image](https://user-images.githubusercontent.com/8158404/57457172-4e2cb580-726f-11e9-9e7a-b66f666caaf3.png)

I **could** use this api endpoint, BUT I don't see to which group the user belongs to. And this information is crucial for my application.

SO, summing up, my issue/question:
How can normal user get other users/user groups?

### Expected behaviour
I want /nextcloud/ocs/v1.php/cloud/groups to work for normal users (at least GET request).

### Actual behaviour
/nextcloud/ocs/v1.php/cloud/groups - is working only for admins.
Nextcloud web interface is using old/unknown official documentation api call /nextcloud/ocs/v1.php/apps/files_sharing/api/v1/sharees .

### Server configuration

**Operating system**: UCS 4.4

**Web server:** ???

**Database:** postgres

**PHP version:** ???

**Nextcloud version:** (see Nextcloud admin page) nextcloud for UCS official application (docker container)

**Updated from an older Nextcloud/ownCloud or fresh install:** no

**Where did you install Nextcloud from:**  UCS official application

**Comment by skjnldsv** [comment]
Well, this one slept through :see_no_evil: 
You can find the appropriate code here: https://github.com/nextcloud/server/blob/fd90af50d910e659aa8df0d380424383c6c09620/apps/files_sharing/src/components/SharingInput.vue#L179 and here https://github.com/nextcloud/server/blob/fd90af50d910e659aa8df0d380424383c6c09620/apps/files_sharing/src/components/SharingInput.vue#L247 for the recommendations
